appengine/standard/storage/appengine-client/main.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START sample]
+"""A sample app that uses GCS client to operate on bucket and file."""
+
+# [START imports]
+import logging
+import os
+
+import cloudstorage as gcs
+from google.appengine.api import app_identity
+
+import webapp2
+
+# [END imports]
+
+# [START retries]
+my_default_retry_params = gcs.RetryParams(
+    initial_delay=0.2, max_delay=5.0, backoff_factor=2, max_retry_period=15)
+gcs.set_default_retry_params(my_default_retry_params)
+# [END retries]
+
+
+class MainPage(webapp2.RequestHandler):
+    """Main page for GCS demo application."""
+
+# [START get_default_bucket]
+    def get(self):
+        bucket_name = os.environ.get(
+            'BUCKET_NAME', app_identity.get_default_gcs_bucket_name())
+
+        self.response.headers['Content-Type'] = 'text/plain'
+        self.response.write(
+            'Demo GCS Application running from Version: ' +
+            os.environ['CURRENT_VERSION_ID'] + '\n')
+        self.response.write('Using bucket name: ' + bucket_name + '\n\n')
+# [END get_default_bucket]
+
+        bucket = '/' + bucket_name
+        filename = bucket + '/demo-testfile'
+        self.tmp_filenames_to_clean_up = []
+
+        try:
+            self.create_file(filename)
+            self.response.write('\n\n')
+
+            self.read_file(filename)
+            self.response.write('\n\n')
+
+            self.stat_file(filename)
+            self.response.write('\n\n')
+
+            self.create_files_for_list_bucket(bucket)
+            self.response.write('\n\n')
+
+            self.list_bucket(bucket)
+            self.response.write('\n\n')
+
+            self.list_bucket_directory_mode(bucket)
+            self.response.write('\n\n')
+
+        except Exception, e:
```

Comment:
Don't catch broad exceptions. In this case, it's better to just not handle errors and let them bubble up to the user.